src/index.ts

```diff
@@ -94,30 +98,43 @@ const addOAuthInterceptor = (
     tokenSecret = null,
     callback = null,
     verifier = null,
-  }: OAuthInterceptorConfig
+  }: Partial<OAuthInterceptorConfig> = {}
```

Comment:
It seems like we lose something here by not requiring the required fields any more.  Maybe this argument could be totally optional and if it's not provided then you MUST provide a per-request config.  I'm a bit doubtful people will really want to provide some "defaults" here and also provide per-request "overrides".  And they could manage that themselves without too much difficulty.